src/main/java/se/sundsvall/notes/service/NoteService.java

```diff
@@ -1,58 +1,58 @@
 package se.sundsvall.notes.service;
 
-import static java.lang.String.format;
-import static java.util.Collections.emptyList;
-import static org.apache.commons.lang3.BooleanUtils.isNotTrue;
-import static org.zalando.problem.Status.NOT_FOUND;
-import static se.sundsvall.notes.service.ServiceConstants.ERROR_NOTE_NOT_FOUND;
-import static se.sundsvall.notes.service.mapper.NoteMapper.toNote;
-import static se.sundsvall.notes.service.mapper.NoteMapper.toNoteEntity;
-import static se.sundsvall.notes.service.mapper.NoteMapper.toNotes;
-
-import java.util.List;
-
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.data.domain.PageRequest;
 import org.springframework.data.domain.Sort;
 import org.springframework.stereotype.Service;
 import org.zalando.problem.Problem;
-
 import se.sundsvall.notes.api.model.CreateNoteRequest;
 import se.sundsvall.notes.api.model.FindNotesRequest;
 import se.sundsvall.notes.api.model.FindNotesResponse;
 import se.sundsvall.notes.api.model.MetaData;
 import se.sundsvall.notes.api.model.Note;
 import se.sundsvall.notes.api.model.UpdateNoteRequest;
 import se.sundsvall.notes.integration.db.NoteRepository;
-import se.sundsvall.notes.integration.db.model.NoteEntity;
+
+import java.util.List;
+
+import static java.lang.String.format;
+import static java.util.Collections.emptyList;
+import static org.apache.commons.lang3.BooleanUtils.isNotTrue;
+import static org.zalando.problem.Status.NOT_FOUND;
+import static se.sundsvall.notes.service.ServiceConstants.ERROR_NOTE_NOT_FOUND;
+import static se.sundsvall.notes.service.mapper.NoteMapper.toNote;
+import static se.sundsvall.notes.service.mapper.NoteMapper.toNoteEntity;
+import static se.sundsvall.notes.service.mapper.NoteMapper.toNotes;
 
 @Service
 public class NoteService {
 
 	@Autowired
 	private NoteRepository noteRepository;
 
-	public String createNote(CreateNoteRequest createNoteRequest) {
-		return noteRepository.save(toNoteEntity(createNoteRequest)).getId();
+	public String createNote(String municipalityId, CreateNoteRequest createNoteRequest) {
+		return noteRepository.save(toNoteEntity(municipalityId, createNoteRequest)).getId();
 	}
 
-	public Note updateNote(String id, UpdateNoteRequest updateNoteRequest) {
-		NoteEntity noteEntity = noteRepository.findById(id)
-			.orElseThrow(() -> Problem.valueOf(NOT_FOUND, format(ERROR_NOTE_NOT_FOUND, id)));
+	public Note updateNote(String id, String municipalityId, UpdateNoteRequest updateNoteRequest) {
+		if (isNotTrue(noteRepository.existsByIdAndMunicipalityId(id, municipalityId))) {
```

Comment:
I don't know if it's better performance to make two calls to db (existsByIdAndMunicipalityId and getReferenceById) instead of just one (getByIdAndMunicipalityId) and keep the .orElseThrow() when the get doesn't return a result to acomplish the logic? My oppinion is that it is more readable to use the .orElseThrow-structure though.